### PR TITLE
[SYCL] Enable native FP add/sub tests for CUDA

### PR DESCRIPTION
**Pull request by Pennycook**
Test for functionality from intel/llvm#3276

Signed-off-by: John Pennycook <john.pennycook@intel.com>

### CODE REVIEW

File: SYCL/AtomicRef/add-native.cpp [hunk 1]
```diff
@@ -1,11 +1,11 @@
 // RUN: %clangxx -fsycl -fsycl-unnamed-lambda -DSYCL_USE_NATIVE_FP_ATOMICS \
 // RUN: -fsycl-targets=%sycl_triple %s -o %t.out
 // RUN: %HOST_RUN_PLACEHOLDER %t.out
-// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out -Xsycl-target-backend=nvptx64-nvidia-cuda-sycldevice --cuda-gpu-arch=sm_60
```
Comment (AGindinson) [inline SYCL/AtomicRef/add-native.cpp]: I believe the arg is being added to the wrong line - the Clang compilation command is what should receive it. The same applies to the `fetch_sub` test.
Meanwhile, the fact that this CUDA-specific arg is passed regardless of the actual GPU target isn't of much bother - it's good that we can rely on the driver to simply ignore it for generic compilations.

Comment (AGindinson) [reply SYCL/AtomicRef/add-native.cpp]: I would also suggest splitting the resulting `RUN` command into more lines so that the format adhered to.

Comment (Pennycook) [reply SYCL/AtomicRef/add-native.cpp]: Thanks for the prompt review, @AGindinson.  Both fixed in 0a01ee8.

**Review by AGindinson — APPROVED** [review]
LGTM!

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
Both tests failed on CUDA. Could you please fix it?

**Comment by Pennycook** [comment]
@vladimirlaz I reran the tests and I see both AtomicRef tests passing.  The only test failing now is Plugin/enqueue-arg-order-image.cpp, which seems to have nothing to do with my changes.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**